src/AutoMapper/MapperConfiguration.cs

```diff
@@ -428,7 +480,20 @@ public LockingConcurrentDictionary(Func<TKey, TValue> valueFactory)
 
         public TValue GetOrAdd(TKey key)
         {
-            return _dictionary.GetOrAdd(key, _valueFactory).Value;
+            try
+            {
+                return _dictionary.GetOrAdd(key, _valueFactory).Value;
+            }
+            catch (InvalidOperationException e) when(e.Message == "ValueFactory attempted to access the Value property of this instance.")
+            {
+                return default(TValue);
+            }
+        }
+
+        public TValue AddOrUpdate(TKey key)
+        {
+            LockingConcurrentDictionary<TKey, TValue> tmpThis = this;
+            return tmpThis._dictionary.AddOrUpdate(key, tmpThis._valueFactory, (tp, mf) => tmpThis._valueFactory(tp)).Value;

```

Comment:
Yea the thing wouldn't let me compile otherwise.  This is what Resharper did to make it compile.